crates/router/src/connector/nmi.rs

```diff
@@ -0,0 +1,641 @@
+        types::ResponseRouterData {
+            response,
+            data: data.clone(),
+            http_code: res.status_code,
+        }
+        .try_into()
+        .change_context(errors::ConnectorError::ResponseHandlingFailed)
+    }
+
+    fn get_error_response(
+        &self,
+        res: types::Response,
+    ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+        self.build_error_response(res)
+    }
+}
+
+impl ConnectorIntegration<api::RSync, types::RefundsData, types::RefundsResponseData> for Nmi {
+    fn get_headers(
+        &self,
+        req: &types::RefundsRouterData<api::RSync>,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
+
+    fn get_url(
+        &self,
+        _req: &types::RefundsRouterData<api::RSync>,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok(format!("{}api/query.php", self.base_url(connectors)))
+    }
+
+    fn get_request_body(
+        &self,
+        req: &types::RefundsRouterData<api::RSync>,
+    ) -> CustomResult<Option<String>, errors::ConnectorError> {
+        let nmi_req = utils::Encode::<nmi::NmiSyncRequest>::convert_and_url_encode(req)
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Some(nmi_req))
+    }
+
+    fn build_request(
+        &self,
+        req: &types::RefundsRouterData<api::RSync>,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        Ok(Some(
+            services::RequestBuilder::new()
+                .method(services::Method::Post)
+                .url(&types::RefundSyncType::get_url(self, req, connectors)?)
+                .headers(types::RefundSyncType::get_headers(self, req, connectors)?)
+                .body(types::RefundSyncType::get_request_body(self, req)?)
+                .build(),
+        ))
+    }
+
+    fn handle_response(
+        &self,
+        data: &types::RefundsRouterData<api::RSync>,
+        res: types::Response,
+    ) -> CustomResult<types::RefundsRouterData<api::RSync>, errors::ConnectorError> {
+        let query_response = String::from_utf8(res.response.to_vec())
+            .into_report()
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        let (transaction_id, condition) = transformers::get_query_info(query_response)?;
+        let refund_status = get_refund_status(condition)?;
+        Ok(types::RefundSyncRouterData {
+            response: Ok(types::RefundsResponseData {
+                connector_refund_id: transaction_id,
+                refund_status,
+            }),
+            ..data.clone()
+        })
+
+        /* To do implement xml deserialization for QueryResponse in PSync and RSync flow
```

Comment:
It was there before and I kept it since no proper documentation exist for NMI and also as a reference if me or someone implement that in future. I can remove it if not needed as it doesn't affect the current functionality.